plugins/modules/packaging/os/portage.py

```diff
@@ -262,7 +266,12 @@ def query_set(module, set, action):
             module.fail_json(msg='set %s cannot be removed' % set)
         return False
 
-    world_sets_path = '/var/lib/portage/world_sets'
+    try:
+        # When providing ROOT via ansible enviroment, portage.root may return a
+        # portage._LegacyGlobalProxy instance. Force it to be a string.
+        world_sets_path = os.path.join(str(portage.root), portage.const.WORLD_SETS_FILE)
```

Comment:
What part of this nested thing can actually cause a `NameError`? It's hard to guess, could you wrap with try-except just that, and move the rest to an else-block?

Reply:
Actually `portage` is the thing that triggers the `NameError`. But it always does, since `portage` is not passed here. This needs to be fixed.